sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -69,7 +69,7 @@ class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
         activeQueryRunIds.synchronized { activeQueryRunIds += s.runId }
         sparkListenerBus.post(s)
         // post to local listeners to trigger callbacks
-        postToAll(s)
+        postToAllSync(s)
```

Comment:
Do you need this change? Seems like it wasn't a synchronous call before. If you need it, then `postToAllSync` isn't really for testing only, and you need to change the command in that method.

Same below.